pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -798,7 +824,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 					}
 
 					pluginParams := &plugin.InputParams{
-						ListenerProtocol:           plugin.ModelProtocolToListenerProtocol(servicePort.Protocol),
+						ListenerProtocol:           plugin.DefaultListenerProtocol(node, servicePort.Protocol),
```

Comment:
This would return `ListenerProtocolAuto` by default. I believe this will break the authentication and authorization support on gateway, see https://github.com/istio/istio/blob/7f03f3f648d15d8926640915f236df49c19932d5/pilot/pkg/networking/plugin/authz/authorization.go#L48 and https://github.com/istio/istio/blob/7f03f3f648d15d8926640915f236df49c19932d5/pilot/pkg/networking/plugin/authn/authentication.go#L47 as they don't currently support `ListenerProtocolAuto`.

Could you double check this? These features are used by a lot customers in production today. We can discuss offline if you want, thank you.

Reply:
The function is only used by sidecar listener not by gateway listener.

Reply:
Discussed offline, thanks for the explanation. To be more specific, this is for sidecar *outbound* listeners which is not used by authN/Z for now.

Reply:
Yes, but why should this be set to auto for everything? What happens to the other plugins that rely on the protocol ?

Reply:
It's not for everything. Only for the last catch all egress listener which doesn't specify port.

Reply:
Adapt authn/authz plugin with protocol sniffing. Check protocol of each filter chain rather than the listener.